src/dotfilepatcher.cpp

```diff
@@ -27,20 +27,146 @@
 
 static const char svgZoomHeader[] =
 "<svg id=\"main\" version=\"1.1\" xmlns=\"http://www.w3.org/2000/svg\" xmlns:xlink=\"http://www.w3.org/1999/xlink\" xml:space=\"preserve\" onload=\"init(evt)\">\n"
+"<!-- removed from style as the pointer interferes with the URL\n"
+".node {cursor: pointer;}\n"
+"-->\n"
 "<style type=\"text/css\"><![CDATA[\n"
+".node, .edge {opacity: 0.7;}\n"
+".node.selected, .edge.selected {opacity: 1;}\n"
 ".edge:hover path { stroke: red; }\n"
 ".edge:hover polygon { stroke: red; fill: red; }\n"
 "]]></style>\n"
 "<script type=\"text/javascript\"><![CDATA[\n"
-"var edges = document.getElementsByTagName('g');\n"
-"if (edges && edges.length) {\n"
+"var elem = document.getElementsByTagName('g');\n"
+"if (elem && elem.length) {\n"
 "  for (var i=0;i<edges.length;i++) {\n"
-"    if (edges[i].id.substr(0,4)=='edge') {\n"
-"      edges[i].setAttribute('class','edge');\n"
+"    if (elem[i].id.substr(0,4)=='edge') {\n"
+"      elem[i].setAttribute('class','edge');\n"
+"      elem[i].setAttribute('class','selected');\n"
+"    }\n"
+"    else if (edges[i].id.substr(0,4)=='Node') {\n"
+"      elem[i].setAttribute('class','node');\n"
+"      elem[i].setAttribute('class','selected');\n"
 "    }\n"
 "  }\n"
 "}\n"
 "]]></script>\n"
+"<script>\n"
+"function addItemToMapOfLists(mapOfLists, name, item) {\n"
+"  // mapOfLists = {\n"
+"  //   'key1': [item1, item2, ...],\n"
+"  //   'key2': [item3, item4, ...],\n"
+"  // }\n"
+"  if (mapOfLists[name].length &gt; 0) {\n"
+"    if (!mapOfLists[name].includes(item)) {\n"
+"      mapOfLists[name].push(item);\n"
+"    }\n"
+"  } else {\n"
+"    mapOfLists[name] = [item];\n"
+"  }\n"
+"}\n"
+"\n"
+"function main() {\n"
```

Comment:
@albert-github It looked like this `main()` function was not really needed, so I removed it. Please let me know if that still breaks something I didn't see.

Reply:
At first glance I don't see a problem.